utils/message-generator/src/main.rs

```diff
@@ -17,87 +17,254 @@ use roles_logic_sv2::{common_messages_sv2::SetupConnectionSuccess, parsers::AnyM
 use std::net::SocketAddr;
 //use serde::{Deserialize as SerdeDeserialize, Serialize as SerdeSerialize};
 use serde_json;
+use tracing::{debug, info};
 
+// TODO RR: Move to `actions.rs`
+/// Represents a valid SV2 type used in the `"value"` key-pair of the `"results"` key value of the
+/// `"actions"` key value. It is used when the `"results"`'s `"type"` is `"match_message_field"`.
+/// For example:
+///    ```
+///    {
+///      "type": "match_message_field",
+///        "value": [
+///          "MiningProtocol",
+///          "OpenStandardMiningChannelSuccess",
+///          "request_id",
+///          { "U32": 89 }
+///        ]
+///     }
+///    ```
+///    where `{ "U32": 89 }` is serialized as `Sv2Type::U32(89)`.
 #[derive(Debug, PartialEq, Eq, Clone, Serialize, Deserialize)]
 enum Sv2Type {
+    /// Message field `bool` value
     Bool(bool),
+    /// Message field `u8` value
     U8(u8),
+    /// Message field `u16` value
     U16(u16),
+    /// Message field `u24` value as a vector of `u8`'s
+    /// TODO: Change variant to `U24Vec(Vec<u8>)`?
     U24(Vec<u8>),
+    /// Message field `u32` value
     U32(u32),
+    /// Message field `u256` value as a vector of `u8`'s
+    /// TODO: Change variant to `U256Vec(Vec<u8>)`?
```

Comment:
@fi3 do we want to change the data type to reflect it is holding a `Vec<u8>` to separate it from the `U16(u16)` variant, for example, that directly stores the value?

Reply:
Dont change

Reply:
TODO: check how used and make sure types are the right size